src/gapi_lwt.ml

```diff
@@ -0,0 +1,21 @@
+let load ~client ~callback =
+  let (wait, wakeup) = Lwt.wait () in
+  let n_callback = function
+    | payload ->
+        Lwt.wakeup wakeup payload ;
+        callback payload
+  in
+  Base.load ~client ~callback:n_callback ;
+  wait
+
+let promise f =
+  let (wait, wakeup) = Lwt.task () in
+  (f (fun x -> Lwt.wakeup wakeup x) : unit) ;
+  wait
```

Comment:
Could you please use this helper function in `load` too?

Reply:
> Could you please use this helper function in `load` too?

I tried but I didn't really understand how I was supposed to to use it in load...

Reply:
```ocaml
let load ~client =
  promise @@ fun wakeup ->
  Base.load ~client ~callback:wakeup
```

Couldn't we write this? (Since [`Lwt.task`](https://ocsigen.org/lwt/latest/api/Lwt#VALtask) and `Lwt.wait` are basically equivalent.)